ocaml/runtime/shared_heap.c

```diff
@@ -958,8 +1018,417 @@ static void compact_update_ephe_list(volatile value *ephe_p)
   }
 }
 
+int compact_count_pools(pool* start_pool) {
+  int count = 0;
+  while( start_pool ) {
+    count++;
+    start_pool = start_pool->next;
+  }
+
+  return count;
+}
+
+/* sort first by chunk_size and then by chunk */
+int compact_compare_pools(const void* v1, const void* v2) {
+  pool* p1 = *(pool**)v1;
+  pool* p2 = *(pool**)v2;
+
+  if (p1->chunk_size < p2->chunk_size) {
+    return 1;
+  } else if (p1->chunk_size > p2->chunk_size) {
+    return -1;
+  } else {
+    if (p1->chunk > p2->chunk) {
+      return -1;
+    } else if (p1->chunk < p2->chunk) {
+      return 1;
+    } else {
+      /* otherwise we want the pools in address order (which helps freeing them later) */
+      return p2 > p1;
+    }
+  }
+}
+
+static atomic_uintnat global_live_pools = ATOMIC_VAR_INIT(0);
```

Comment:
Since this is now only used in `compact_phase_two_mark`, which is run by a single domain while other domains are blocked, and which also computes the same value (it's the value of _pools_count_ at line 1203), I think we should lose the global and the various atomic updates to it, and just have it as a local variable in that function.

Reply:
Yea, done.